icu4c/source/test/intltest/alphaindextst.cpp

```diff
@@ -756,4 +757,52 @@ void AlphabeticIndexTest::checkHasBuckets(const Locale &locale, UScriptCode scri
             uscript_getScript(bucket->getLabel().char32At(0), errorCode));
 }
 
+void AlphabeticIndexTest::checkMaxLabelCount() {
+    IcuTestErrorCode errorCode(*this, "checkMaxLabelCount");
+    UErrorCode status = U_ZERO_ERROR;
+    Locale locale("en");
+    UnicodeString loc = locale.getName();
+    AlphabeticIndex aindex(locale, status);
+
+    int32_t defaultMax = aindex.getMaxLabelCount();
+    assertEquals("Expect max count = 99", 99, defaultMax);
+
+    aindex.setMaxLabelCount(237, status);
+    if (U_FAILURE(errorCode)) {
+      dataerrln("%s %d  Error in index setMaxLabelCount",  __FILE__, __LINE__);
+      return;
+    }
+
+    // Verify that all bucket are cleared after setting max bucket count.
+    const UnicodeString bucketLabel = aindex.getBucketLabel();
+    assertEquals("Expect empty string", "", bucketLabel);
+
+    int32_t maxChanged = aindex.getMaxLabelCount();
+    assertEquals("Expect max count = 237", 237, maxChanged);
+
+    int32_t bigMax = 100000;
+    aindex.setMaxLabelCount(bigMax, errorCode);
+    if (U_FAILURE(errorCode)) {
+      dataerrln("%s %d  Error in index setMaxLabelCount to bigMax",
+                __FILE__, __LINE__);
+      return;
+    }
+    maxChanged = aindex.getMaxLabelCount();
+    assertEquals("Expect max count = 100000", bigMax, maxChanged);
+
+    int32_t numBuckets = aindex.getBucketCount(status);
+    int32_t expectedBuckets = 0x1C;
+    assertEquals("Expect numBuckets = 1", expectedBuckets, numBuckets);
```

Comment:
You are not actually expecting it to be 1. I suggest you don't hardcode the expected value in the message, since they are easily out of sync.

Reply:
I'm fond of using just the test framework macro WHERE for expect messages. It gives the file name and line number, which makes it much easier to zero in on a failing test, compared with searching for the text of the error message in  the test source code. Plus it's quick and easy to write; it's all you really need.